packages/plugin-constraints/sources/commands/constraints/query.ts

```diff
@@ -0,0 +1,45 @@
+import {Configuration, Project, PluginConfiguration} from '@berry/core';
+import {Writable}                                    from 'stream';
+
+import {Constraints}                                 from '../../Constraints';
+
+function valueToString(value: string|null): string {
+  if (typeof value !== `string`)
+    return `${value}`;
+
+  if (value.match(/^[a-zA-Z][a-zA-Z0-9_]+$/))
+    return value;
+
+  return `'${value}'`;
+}
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: any, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`constraints query <query>`)
+
+  .categorize(`Constraints-related commands`)
+  .describe(`query the constraints`)
+
+  .detail(`
+    This command will output all matches to the given prolog query
+  `)
+
+  .example(
+    `Find all fixable invalid dependencies`,
+    `yarn constraints query 'workspace(WorkspaceCwd), dependency_type(DependencyType), gen_enforced_dependency_range(WorkspaceCwd, DependencyName, DependencyRange, DependencyType).'`,
```

Comment:
Simpler example maybe, to keep it on one line? For example:

> Find all dependencies of all workspaces

Reply:
Can do! I used this example because it's one I'd use to test the constraints, but adding an example like "List all direct dependencies throughout the workspace" `yarn constraints query 'workspace_has_dependency(_, Dependency, _, _).'` would make for a shorter and easier example

Reply:
I ran that command and noticed that it logs undefined values for `_`. Weirdly I didn't see this before. I'll filter those values out as `_` implies the value is ignored